### PR TITLE
ssl: Set a global boolean to enable SSL across Minio

### PR DESCRIPTION
**Pull request by harshavardhana**
## Description
We have been using `isSSL()` everywhere we can set
a global value once and re-use it again.

## Motivation and Context
Cleanup and some simplification.

## How Has This Been Tested?
Manually using local certs.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: cmd/server-main.go [hunk 1]
```diff
@@ -317,38 +317,55 @@ func isAnyEndpointLocal(eps []*url.URL) bool {
...
+		return "", "", fmt.Errorf("Port cannot be empty or '0', please use `--address` to pick a specific port")
+	}
+
+	// Check if port is available.
+	if err = checkPortAvailability(port); err != nil {
+		return "", "", err
+	}
+
+	// Parse port.
+	if _, err = strconv.Atoi(port); err != nil {
```
Comment (krisis) [inline cmd/server-main.go]: we should parse port before checking port availability.

File: cmd/server-main_test.go [hunk 2]
```diff
@@ -63,20 +63,66 @@ func TestGetListenIPs(t *testing.T) {
...
+	}{
+		// Test 1 - successful.
+		{
+			addr: ":" + getFreePort(),
+			err:  nil,
+		},
+		// Test 2 port empty.
+		{
+			addr: ":0",
+			err:  errors.New("Port cannot be empty or '0', please use `--address` to pick a specific port"),
```
Comment (bweston92) [inline cmd/server-main_test.go]: @harshavardhana should we not define these somewhere so we can just reference to them?

main.go
```
var errEmptyPortSpecified = errors.New("Port cannot be empty or '0', please use `--address` to pick a specific port")

...
return errEmptyPortSpecified
...
```

main_test.go
```
{
  addr: ":" + getFreePort(),
  err: errEmptyPortSpecified,
}
```

**Review by krisis — CHANGES_REQUESTED** [review]

File: cmd/server-main.go (hunk 1 above)
Comment (harshavardhana) [reply cmd/server-main.go]: Oh yes that is correct.

File: cmd/server-main_test.go (hunk 2 above)
Comment (harshavardhana) [reply cmd/server-main_test.go]: We could define but since they are not used elsewhere it's not defined in that manner, but not a bad idea either.

Comment (bweston92) [reply cmd/server-main_test.go]: Ok will leave it in your hands :)

**Review by vadmeste — APPROVED** [review]

**Pull request merged**